app/flatpak-builtins-config.c

```diff
@@ -63,18 +63,66 @@ looks_like_a_language (const char *s)
   return TRUE;
 }
 
+static gboolean
+looks_like_a_locale (const char *s)
+{
+  gchar **locale = g_strsplit (s, "_", 3);
+  int i;
+  int len;
+
+  len = strlen (locale[0]);
+  if (len < 2 || len > 3)
+    return FALSE;
+
+  for (i = 0; i < len; i++)
+    {
+      if (!g_ascii_isalpha (locale[0][i]) || !g_ascii_islower (locale[0][i]))
+        return FALSE;
+    }
+
+  if (locale[1] != NULL)
+    {
+      len = strlen (locale[1]);
+
+      if (len < 2 || len > 4)
+        return FALSE;
+
+      for (i = 0; i < len; i++)
+        {
+          if (!g_ascii_isalpha (locale[1][i]))
+            return FALSE;
+        }
+    }
+
+  if (locale[2] != NULL)
+    {
+      len = strlen (locale[2]);
+
+      if (len < 2 || len > 3)
+        return FALSE;
+
+      for (i = 0; i < len; i++)
+        {
+          if (!g_ascii_isalpha (locale[2][i]) || !g_ascii_isupper (locale[2][i]))
+            return FALSE;
+        }
+    }
+
+  return TRUE;
+}
+
 static char *
-parse_lang (const char *value, GError **error)
+parse_locale (const char *value, GError **error)
 {
   g_auto(GStrv) strs = NULL;
   int i;
 
   strs = g_strsplit (value, ";", 0);
   for (i = 0; strs[i]; i++)
     {
-      if (!looks_like_a_language (strs[i]))
+      if (!looks_like_a_locale (strs[i]))
```

Comment:
I think this should be 

```
if (!looks_like_a_language () && !looks_like_a_locale ())
...
```
Since I assume we want to accept both language codes and locale identifiers in extra-languages.

Reply:
Yes! but since `looks_like_a_locale` already checks if the first bit before `_` is a valid lowercase string, I don't think we need to match against both